### PR TITLE
Sklearn autologging bug fix: Don't compute metrics when labels are missing

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

All of the metrics that are currently computed as part of sklearn autologging require labels to be specified during training (i.e. they require a `y` argument to be passed to `fit()`. Many unsupervised algorithms don't require `y`. For these algorithms, metric computation currently fails and prints an "unexpected error" warning message:

```
import mlflow
mlflow.sklearn.autolog()

import sklearn.datasets
import sklearn.cluster

def get_iris():
    iris = sklearn.datasets.load_iris()
    X = iris.data[:, :2]
    y = iris.target
    return X, y

X, y = get_iris()
sklearn.cluster.KMeans().fit(X=X)
```

```
2021/05/06 17:50:58 INFO mlflow.utils.autologging_utils: Created MLflow autologging run with ID '51267604cc9c4159b54f0b540bf0b66a', which will track hyperparameters, performance metrics, model artifacts, and lineage information for the current sklearn workflow
2021/05/06 17:50:58 WARNING mlflow.utils.autologging_utils: Encountered unexpected error during sklearn autologging: 'y'
```

This PR fixes the issue by handling the case where `y` is not specified.

## How is this patch tested?

Unit tests

## Release Notes

Fix a bug where an "unexpected error" warning message was printed during unsupervised model training with scikit-learn autologging enabled.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/sklearn/__init__.py [hunk 1]
```diff
@@ -756,14 +756,20 @@ def infer_model_signature(input_example):
...
+        logged_metrics = _log_estimator_content(
             estimator=estimator,
             prefix=_TRAINING_PREFIX,
             run_id=mlflow.active_run().info.run_id,
             X=X,
             y_true=y_true,
             sample_weight=sample_weight,
         )
+        if y_true is None and not logged_metrics:
+            _logger.warning(
```
Comment (dbczumar) [inline mlflow/sklearn/__init__.py:768]: In the future, we can likely add metrics for unsupervised techniques that don't depend on labels (e.g. https://scikit-learn.org/stable/modules/classes.html#pairwise-metrics for clustering). For now, printing a warning seems like a good idea, since users *can* pass labels to clustering methods and record metrics.

**Review by jinzhang21 — APPROVED** [review]
LGTM.

**Comment by dbczumar** [comment]
Failing tests are a result of a PyTorch Lightning breaking change released about an hour ago: https://pypi.org/project/pytorch-lightning/. They are unrelated to this PR. Merging.
